crypto/ec/ec_key.c

```diff
@@ -517,6 +519,34 @@ int ec_key_private_check(const EC_KEY *eckey)
     return 1;
 }
 
+#ifndef OPENSSL_NO_SM2
+int sm2_key_private_check(const EC_KEY *eckey)
+{
+    int ret = 0;
+    BIGNUM *max = NULL;
+
+    if (eckey == NULL || eckey->group == NULL || eckey->priv_key == NULL) {
+        ECerr(0, ERR_R_PASSED_NULL_PARAMETER);
+        return 0;
+    }
+
+    /* range of SM2 private key is [1, n-1) */
```

Comment:
]?

Reply:
This is intentional, `1` is included in the range, `n-1` is excluded from the range.